core/node/node_framework/src/service/tests.rs

```diff
@@ -0,0 +1,235 @@
+use std::sync::Arc;
+
+use anyhow::anyhow;
+use futures::lock::Mutex;
```

Comment:
You can use mutexes from `std` to avoid dealing with multiple tokio Runtimes, there is no need for async mutexes here.